functions/src/business-logics/share-logics.ts

```diff
@@ -1,4 +1,80 @@
+import {Action, LogicConfig, LogicResultDoc} from "emberflow/lib/types";
+import {Entity} from "../db-structure";
+import {db} from "emberflow/lib";
 // TODO write onShareLogic
 //  copy post to user's timeline
 //  increment post sharesCount
 //  create a notification for post's author
+
+
+export const onShareLogic: LogicConfig = {
+  name: "onShareLogic",
+  actionTypes: ["create"],
+  modifiedFields: ["all"],
+  entities: [Entity.Share],
+  logicFn: async (action: Action) => {
+    const {
+      eventContext: {
+        docPath,
+      },
+      user,
+    } = action;
+    const shareDocRef = db.doc(docPath);
+    const postDocRef = shareDocRef.parent.parent;
+
+    if (!postDocRef) {
+      return {
+        name: "onShareLogic",
+        status: "error",
+        message: "Post not found",
+        documents: [],
+      };
+    }
+
+    const postDoc = await postDocRef.get();
+    const post = postDoc.data();
+    if (post === undefined) {
+      return {
+        name: "onShareLogic",
+        status: "error",
+        message: "Post not found",
+        documents: [],
+      };
+    }
+
+    const {
+      "@id": postId,
+      createdBy: {
+        "@id": postAuthorId,
+      },
+    } = post;
+    const timelineLogicResultDoc: LogicResultDoc = {
+      action: "copy",
+      dstPath: `users/${user["@id"]}/timeline/${postId}`,
+      srcPath: postDocRef.path,
+    };
+    const postLogicResultDoc: LogicResultDoc = {
+      action: "create",
+      dstPath: docPath,
```

Comment:
the value of docPath here is "posts/postId/shares/shareId" because you are using the share entity not post. if you want to use the post path you can get it in the postDocRef